doc/api/esm.md

```diff
@@ -286,13 +282,18 @@ const { default: barData } =
   await import('./bar.json', { with: { type: 'json' } });
 ```
 
-Node.js supports the following `type` values, for which the attribute is
-mandatory:
+Node.js only supports the `type` attribute, for which it supports the following values:
 
 | Attribute `type` | Needed for       |
 | ---------------- | ---------------- |
 | `'json'`         | [JSON modules][] |
 
+The `type: 'json'` attribute is mandatory when importing JSON modules.
+
+> This feature was previously named "Import assertions", and  it used the
+> `assert` keyword instead of `with`. Any uses in code of the prior `assert`
+> keyword should be updated to use `with` instead.
```

Comment:
```suggestion
> This feature was previously named "Import assertions", and it used the
> `assert` keyword instead of `with`. Any uses in code of the prior `assert`
> keyword should be updated to use `with` instead.
```
but tbh, I'm not sure it should be in Node.js docs at all, the runtime warning (or straight up syntax error in recent versions) should be enough IMO
```suggestion
```

Reply:
Agree :+1: